code/drasil-code/lib/Language/Drasil/Code/Imperative/GOOL/ClassInterface.hs

```diff
@@ -66,5 +66,6 @@ data ReadMeInfo = ReadMeInfo {
   configFP :: [FilePath],
   caseName :: CaseName,
   examplePurpose :: ExamplePurpose,
-  exampleDescr :: ExampleDescr
+  exampleDescr :: ExampleDescr,
+  sampleFile :: FilePath
```

Comment:
I'm not sure I like this: do all `README`s have a 'sample File' associated to them? If not, then we shouldn't require this information for all of them.

For optional information, it should be wrapped in a `Maybe`. We shouldn't use that `FilePath` is implemented as `List` to fake "not present" via an empty list. That is too fragile.

Reply:
The `sampleFile` is just referring to the sample input files that are present with all the `ReadMe`s in all examples. Thus, my code was more based on obtaining the `FilePath` from the input files and extracting all the information from those `FilePaths`s. 

I tried obtaining the `FilePath` for the `ReadMe`s but since all of those were obtained using the` IO ()` methods such as `getCurrentDirectory`, etc. I was unable to obtain the `String`/`FilePath` from them in a manner that would not include me using `IO ()` as one of the types throughout the entire fix.

Another potential way that I tried to solve this issue was to create a new choice within the `Choices `area which would take in an integer value which would have a default value of 4. This value essentially would obtain the number of folders that we would have had to go back to in order to get the image. We could then obtain any changes to that value for each example and then generate the image as well. That would I believe remove the modifications done to the Maybe Filepath as our code would no longer be based on the `FilePath` of the input file. 

Let me know if you would like me to modify it such that we create the new `Choice`.

Reply:
Update: The method that I am suggesting in the above comment can be viewed [here ](https://github.com/JacquesCarette/Drasil/tree/3418-Fix-Drasil-Logo-Path-on-READMEs-alt)on a different branch.